exporter/awsxrayexporter/internal/translator/segment.go

```diff
@@ -60,16 +61,173 @@ const (
+	segments, err := MakeSegmentsFromSpan(span, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+	if err == nil {
+		var documents []string
+
+		for _, v := range segments {
+			document, documentErr := MakeDocumentFromSegment(v)
+			if documentErr != nil {
+				return nil, documentErr
+			}
+
+			documents = append(documents, document)
+		}
+
+		return documents, nil
+	}
+
+	return nil, err
+}
+
+// MakeSegmentsFromSpan creates one or more segments from a span
+func MakeSegmentsFromSpan(span ptrace.Span, resource pcommon.Resource, indexedAttrs []string, indexAllAttrs bool, logGroupNames []string, skipTimestampValidation bool) ([]*awsxray.Segment, error) {
+	var isLocalRoot = false
+
+	if myAwsSpanKind, ok := span.Attributes().Get(awsSpanKind); ok {
+		if localRoot == myAwsSpanKind.Str() {
+			isLocalRoot = true
+		}
+	}
+
+	if !isLocalRoot {
+		segment, err := MakeSegment(span, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return []*awsxray.Segment{segment}, nil
+	}
+
+	var hasDependency = (span.Kind() != ptrace.SpanKindServer) && (span.Kind() != ptrace.SpanKindInternal)
+	var segments []*awsxray.Segment
+
+	var serviceSegmentID = newSegmentID()
+
+	// If it is a dependency, then we need to create a subsegment
+	if hasDependency {
+		var dependencySpan ptrace.Span = ptrace.NewSpan()
+		span.CopyTo(dependencySpan)
+
+		dependencySpan.SetParentSpanID(serviceSegmentID)
+
+		dependencySubsegment, err := MakeSegment(dependencySpan, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+		if err != nil {
+			return nil, err
+		}
+
+		// Set the name
+		if myAwsRemoteService, ok := span.Attributes().Get(awsRemoteService); ok {
+			dependencySubsegment.Name = awsxray.String(myAwsRemoteService.Str())
+		}
+
+		// Make this a subsegment
+		dependencySubsegment.Type = awsxray.String("subsegment")
+
+		// Remove span links from consumer spans
+		if span.Kind() == ptrace.SpanKindConsumer {
+			dependencySubsegment.Links = nil
+		}
+
+		segments = append(segments, dependencySubsegment)
+
+		// We always create a segment for the service
+		var serviceSpan ptrace.Span = ptrace.NewSpan()
+		span.CopyTo(serviceSpan)
+
+		// Set the span id to the one internally generated
+		serviceSpan.SetSpanID(serviceSegmentID)
+
+		removeAnnotations := []string{
+			"aws.remote.service",
+			"aws.remote.operation",
+			"remote.target",
+			"K8s.RemoteNamespace",
+		}
+
+		for _, v := range removeAnnotations {
+			serviceSpan.Attributes().Remove(v)
+		}
+
+		serviceSegment, err := MakeSegment(serviceSpan, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+		// Set the name
+		if myAwsLocalService, ok := span.Attributes().Get(awsLocalService); ok {
+			serviceSegment.Name = awsxray.String(myAwsLocalService.Str())
+		}
```

Comment:
I am not sure this worked for me - I was getting the Service node named after the original SpanName when I was experimenting with this.